src/device-state/current-state.ts

```diff
@@ -165,7 +156,7 @@ const backoff = (
 	setTimeout(() => retry(attempts + 1), delay);
 };
 
-function reportCurrentState(attempts = 0) {
+function reportCurrentState(reporter: any, attempts = 0) {
```

Comment:
Unsure how to make reporter have the correct typings so that it is a throttled report function shown in https://github.com/balena-os/balena-supervisor/pull/1885/files#diff-4201a9052aedc0dfdf3d8eb0566ecc3ac822602e1a872e858cddecb2ac772b8cR257-R261

Reply:
It'd be:
```suggestion
function reportCurrentState(reporter: (opts: StateReport, resolve: () => void, reject: (e: Error) => void) => void, attempts = 0) {
```
but if you wanted to infer it then it'd be
```ts
function reportCurrentState(reporter:(...args: Parameters<typeof $fn>) => void, attempts = 0) {
```
Either way it wouldn't enforce that it's throttled but merely that it looks like throttled function (but that's pretty much function types in general, it takes/returns the right types and hopefully does the right thing to transition between them)